bot/exts/evergreen/wikipedia.py

```diff
@@ -46,37 +47,59 @@ async def wiki_request(self, channel: TextChannel, search: str) -> Optional[List
-
-                if number_of_results:
-                    results = raw_data["query"]["search"]
-                    lines = []
-
-                    for article in results:
-                        line = WIKI_SEARCH_RESULT.format(
-                            name=article["title"],
-                            description=unescape(
-                                re.sub(
-                                    WIKI_SNIPPET_REGEX, "", article["snippet"]
-                                )
-                            ),
-                            url=f"https://en.wikipedia.org/?curid={article['pageid']}"
-                        )
-                        lines.append(line)
-
-                    return lines
-
-                else:
-                    await channel.send(
-                        "Sorry, we could not find a wikipedia article using that search term."
+            if (status := resp.status) != 200:
+                log.info(f"Unexpected response `{resp.status}` while searching wikipedia for `{search}`")
+                raise ExternalAPIError("Wikipedia API", status)
+            raw_data = await resp.json()
+            number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
+
+            if number_of_results:
+                results = raw_data["query"]["search"]
+                lines = []
+
+                for article in results:
+                    line = WIKI_SEARCH_RESULT.format(
+                        name=article["title"],
+                        description=unescape(
+                            re.sub(
+                                WIKI_SNIPPET_REGEX, "", article["snippet"]
+                            )
+                        ),
+                        url=f"https://en.wikipedia.org/?curid={article['pageid']}"
+                    )
+                    lines.append(line)
+
+                return lines
+
+            raw_data = await resp.json()
+            if not raw_data.get("query", None):
+                if error := raw_data.get("errors"):
+                    log.error(f"There was an error while communicating with the Wikipedia API: {error}")
+                raise ExternalAPIError("Wikipedia API", status, error)
+
+            number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
+
+            if number_of_results:
+                results = raw_data["query"]["search"]
+                lines = []
+
+                for article in results:
+                    line = WIKI_SEARCH_RESULT.format(
+                        name=article["title"],
+                        description=unescape(
+                            re.sub(
+                                WIKI_SNIPPET_REGEX, "", article["snippet"]
+                            )
+                        ),
+                        url=f"https://en.wikipedia.org/?curid={article['pageid']}"
                     )
-                    return
+                    lines.append(line)
+
+                return lines
```

Comment:
Looks like when you where resolving git conflicts, you merged both the changes. Kindly remove one.